### ISSUE TITLE
bug: misconfig leading to non-graceful crash

### ISSUE THREAD

**Issue by ableuler** [rancher/terraform-provider-rke#301]
I've had a misconfiguration in my terraform setup, leading to a fatal crash of the rke plugin when running `terraform plan`.

- Terraform v1.0.1 on darwin_amd64
- provider registry.terraform.io/rancher/rke v1.2.3

Please let me know if there's any useful information I can provide.

Relevant logs section:
```
[...]
module.rancher-cluster.rke_cluster.cluster: Refreshing state... [id=8a141857-2c36-42c5-830b-121462103e5f]
2021-06-29T17:41:27.458+0200 [TRACE] GRPCProvider: ReadResource
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: PANIC: runtime error: invalid memory address or nil pointer dereference
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: [signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x26b9bcf]
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3:
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: goroutine 34 [running]:
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: github.com/rancher/terraform-provider-rke/rke.readClusterState(0xc000541180, 0x0, 0x0, 0x0)
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: 	/go/src/github.com/rancher/terraform-provider-rke/rke/resource_rke_cluster.go:327 +0x21f
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: github.com/rancher/terraform-provider-rke/rke.resourceRKEClusterRead(0xc000541180, 0x290a560, 0xc0005793b0, 0xc000541180, 0x0)
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: 	/go/src/github.com/rancher/terraform-provider-rke/rke/resource_rke_cluster.go:115 +0xe3
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc001100990, 0xc00007d7c0, 0x290a560, 0xc0005793b0, 0xc000330740, 0x0, 0x0)
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: 	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/helper/schema/resource.go:460 +0x119
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc0002f81d8, 0x2fd1540, 0xc000d18090, 0xc0008c0060, 0xc0002f81d8, 0xc000d18090, 0xc00079cb78)
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: 	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/internal/helper/plugin/grpc_provider.go:525 +0x3d8
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x2afa2c0, 0xc0002f81d8, 0x2fd1540, 0xc000d18090, 0xc0008c0000, 0x0, 0x2fd1540, 0xc000d18090, 0xc000d80000, 0x3071f)
2021-06-29T17:41:27.481+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: 	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/internal/tfplugin5/tfplugin5.pb.go:3269 +0x217
2021-06-29T17:41:27.482+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000082c00, 0x2fed820, 0xc0005a4a80, 0xc000292700, 0xc000578540, 0x3fd4470, 0x0, 0x0, 0x0)
2021-06-29T17:41:27.482+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: 	/go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:1024 +0x501
2021-06-29T17:41:27.482+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: google.golang.org/grpc.(*Server).handleStream(0xc000082c00, 0x2fed820, 0xc0005a4a80, 0xc000292700, 0x0)
2021-06-29T17:41:27.482+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: 	/go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:1313 +0xd3f
2021-06-29T17:41:27.482+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0005d8bf0, 0xc000082c00, 0x2fed820, 0xc0005a4a80, 0xc000292700)
2021-06-29T17:41:27.482+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: 	/go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:722 +0xa1
2021-06-29T17:41:27.482+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-06-29T17:41:27.482+0200 [DEBUG] provider.terraform-provider-rke_v1.2.3: 	/go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:720 +0xa1
2021-06-29T17:41:27.484+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-06-29T17:41:27.484+0200 [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Unavailable desc = transport is closing"
2021-06-29T17:41:27.484+0200 [TRACE] vertex "module.rancher-cluster.rke_cluster.cluster": visit complete
2021-06-29T17:41:27.484+0200 [TRACE] vertex "module.rancher-cluster.rke_cluster.cluster": dynamic subgraph encountered errors: Plugin did not respond
2021-06-29T17:41:27.484+0200 [TRACE] vertex "module.rancher-cluster.rke_cluster.cluster": visit complete
2021-06-29T17:41:27.484+0200 [TRACE] vertex "module.rancher-cluster.rke_cluster.cluster (expand)": dynamic subgraph encountered errors: Plugin did not respond
2021-06-29T17:41:27.484+0200 [TRACE] vertex "module.rancher-cluster.rke_cluster.cluster (expand)": visit complete
2021-06-29T17:41:27.485+0200 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/rancher/rke/1.2.3/darwin_amd64/terraform-provider-rke_v1.2.3 pid=13295 error="exit status 2"
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/rancher/rke\"].default (close)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-cluster.output.cluster (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-cluster.local_file.kube_cluster_yaml (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/gavinbunney/kubectl\"].rancher" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/hashicorp/local\"] (close)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-cluster.null_resource.wait_drain (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/hashicorp/helm\"].rancher" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-cluster.output.dependency (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/rancher/rancher2\"].bootstrap" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-cluster (close)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.var.rancher_hostname (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.var.kubeconfig_path (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.var.password (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.var.kubeconfig_path (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.var.cluster_green_light (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.var.cluster_green_light (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.module.namespace (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.null_resource.wait_helm_delete_cert_manager (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.module.namespace.var.kubeconfig_path (expand)" errored, so skipping
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.module.namespace.var.namespace (expand)" errored, so skipping

Stack trace from the terraform-provider-rke_v1.2.3 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x26b9bcf]

goroutine 34 [running]:
github.com/rancher/terraform-provider-rke/rke.readClusterState(0xc000541180, 0x0, 0x0, 0x0)
	/go/src/github.com/rancher/terraform-provider-rke/rke/resource_rke_cluster.go:327 +0x21f
github.com/rancher/terraform-provider-rke/rke.resourceRKEClusterRead(0xc000541180, 0x290a560, 0xc0005793b0, 0xc000541180, 0x0)
	/go/src/github.com/rancher/terraform-provider-rke/rke/resource_rke_cluster.go:115 +0xe3
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc001100990, 0xc00007d7c0, 0x290a560, 0xc002021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.module.namespace.local.manifest (expand)" errored, so skipping
05793b0, 0xc000330740, 0x0, 0x0)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/helper/schema/resource.go:460 +0x119
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc0002f81d8, 0x2fd1540, 0xc000d18090, 0xc0008c0060, 0xc0002f81d8, 0xc000d18090, 0xc00079cb78)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/internal/helper/plugin/grpc_provider.go:525 +0x3d8
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x2afa2c0, 0xc0002f81d8, 0x2fd1540, 0xc000d18090, 0xc0008c0000, 0x0, 0x2fd1540, 0xc000d18090, 0xc000d80000, 0x3071f)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/internal/tfplugin5/tfplugin5.pb.go:3269 +0x217
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000082c00, 0x2fed820, 0xc0005a4a80, 0xc000292700, 0xc000578540, 0x3fd4470, 0x0, 0x0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:1024 +0x501
google.golang.org/grpc.(*Server).handleStream(0xc000082c00, 0x2fed820, 0xc0005a4a80, 0xc000292700, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:1313 +0xd3f
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0005d8bf0, 0xc000082c00, 0x2fed820, 0xc0005a4a80, 0xc000292700)
	/go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:722 +0xa1
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:720 +0xa1

Error: The terraform-provider-rke_v1.2.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
2021-06-29T17:41:27.485+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.module.namespace.null_resource.namespace (expand)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.module.namespace.output.dependency (expand)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.module.namespace (close)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/hashicorp/null\"] (close)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.helm_release.cert-manager (expand)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.kubectl_manifest.cluster-issuer (expand)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/gavinbunney/kubectl\"].rancher (close)" errored, so skipping
2021-06-29T17:41:
27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager.output.dependency (expand)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.module.cert-manager (close)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.helm_release.rancher (expand)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/hashicorp/helm\"].rancher (close)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.rancher2_bootstrap.admin (expand)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment.output.credentials (expand)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/rancher/rancher2\"].admin" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/rancher/rancher2\"].bootstrap (close)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-authentication.rancher2_user.local (expand)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-authentication.rancher2_auth_config_github.github_oauth (expand)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "output.credentials" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-authentication.rancher2_global_role_binding.admin (expand)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-deployment (close)" errored, so skipping
2021-06-29T17:41:27.486+0200 [TRACE] dag/walk: upstream of "module.rancher-authentication (close)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "module.setup-cluster-alerts (expand)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "module.setup-cluster-alerts.var.cluster_name (expand)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "module.setup-cluster-alerts.var.notifier_slack_webhook (expand)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "module.setup-cluster-alerts.var.notifier_slack_channel (expand)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "module.setup-cluster-alerts.data.rancher2_cluster.imported-cluster (expand)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "module.setup-cluster-alerts.rancher2_notifier.slack_notification (expand)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "module.setup-cluster-alerts.rancher2_cluster_alert_group.cluster_healthy (expand)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "module.setup-cluster-alerts.rancher2_cluster_alert_rule.workers_ready (expand)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "module.setup-cluster-alerts.rancher2_cluster_alert_rule.etcd_available (expand)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "module.setup-cluster-alerts.rancher2_cluster_alert_rule.controlplane_ready (expand)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "module.setup-cluster-alerts (close)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/rancher/rancher2\"].admin (close)" errored, so skipping
2021-06-29T17:41:27.487+0200 [TRACE] dag/walk: upstream of "root" errored, so skipping
2021-06-29T17:41:27.487+0200 [INFO]  backend/local: plan operation completed
╷
│ Error: Plugin did not respond
│
│   with module.rancher-cluster.rke_cluster.cluster,
│   on ../../../modules/rke/rke.tf line 26, in resource "rke_cluster" "cluster":
│   26: resource "rke_cluster" "cluster" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
2021-06-29T17:41:27.500+0200 [DEBUG] provider: plugin exited
2021-06-29T17:41:27.502+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-06-29T17:41:27.502+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-06-29T17:41:27.503+0200 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/null/3.1.0/darwin_amd64/terraform-provider-null_v3.1.0_x5 pid=13292
2021-06-29T17:41:27.503+0200 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/local/2.1.0/darwin_amd64/terraform-provider-local_v2.1.0_x5 pid=13291
2021-06-29T17:41:27.503+0200 [DEBUG] provider: plugin exited
2021-06-29T17:41:27.503+0200 [DEBUG] provider: plugin exited
```

**Comment by rawmind0** [comment]
Submitted PR https://github.com/rancher/terraform-provider-rke/pull/302 to fix the issue

**Comment by rawmind0** [comment]
PR #302 is already merged. The fix will be included on next tfp release, v1.2.4
